### ISSUE TITLE
How do I create a pagination in a layout?

### ISSUE THREAD

**Issue by adnanlah** [11ty/eleventy#1337]
Hello,
In my app I have a categories collection, they all share the same layout "category.njk". I want each category to be paginated with pages of products just from the layout.
I started with this front matter in my category layout:

```
---
layout: "base.njk"
pagination
 data: collections[name]
 size: 5
permalink: /category/{{name}}/{{pagination.pageNumber}}/index.html
---
<div class="text-3xl mb-3 text-white">Category name: {{name}}</div>
{% set products = pagination.items %}
	<div class="col-span-4">
		<div class="grid grid-cols-4 gap-4 mb-6">
		{% include "productslinks.njk" %}
		</div>
		{% include "pagination.njk" %}
	</div>
</div>
```
I tried to get the collections based on the category's name but it didn't work, it gives me an error about resolving the data value. I think that's because the program interprets collections[name] as a string!

Then I tried the JS front matter:
```
---js
{
layout: "base.njk",
pagination: {
 data: "collections.products",
 size: 5,
 before: function(data) {
  return data.filter(entry => entry.data.tags.includes(name));
 }
},
permalink: "/category/{{name}}/{{pagination.pageNumber}}/index.html"
}
---
```
and it didn't work either as I couldn't get the value name of the category at all.

Another I thought about is filter but I can't make them dynamic, in case a user added a new category so that means it needs its filter and that's impossible.
Here is a sample of one of the categories:
```
---
layout: category.njk
name: motherboards
tags: components
---
```
And this is my app structure:
http://prntscr.com/truyw5

In other words how can I filter a collection before I set it to the pagination data? Or how can I get dynamic collections in the front matter, because in the markdown I can easilly get a dynamic collection like `collection[aVariable]` but I can't in the front matter?

I hope my question is understandable and thank you!
